src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -64,15 +65,55 @@ public class QueryService : IQueryService
             return new QueryResourceResponse(Array.Empty<DicomDataset>(), queryExpression.ErroneousTags);
         }
 
-        IEnumerable<DicomDataset> instanceMetadata = await Task.WhenAll(
-            queryResult.DicomInstances.Select(x => _metadataStore.GetInstanceMetadataAsync(x, cancellationToken)));
-
         var responseBuilder = new QueryResponseBuilder(queryExpression);
-        IEnumerable<DicomDataset> responseMetadata = instanceMetadata.Select(m => responseBuilder.GenerateResponseDataset(m));
+        IEnumerable<DicomDataset> responseMetadata = null;
+
+        if (QueryLimit.ContainsComputedTag(queryExpression.IELevel, queryExpression.IncludeFields.DicomTags))
```

Comment:
If in case, I make a study level query and asks for NumberOfSeriesInstances, isn't  this code it will return the result without that column? don't we need to throw an exception?

Reply:
Today we do not throw for any includefield. We just validate it a valid dicomtag. They can indlude any field today.
If we can serve it we will, otherwise we ignore it.
They get Modality at study level query today.